### PR TITLE
Add type casts for integer and enum types

### PR DESCRIPTION
**Pull request by AgathiyanB**
## Description

Part of #7931. Sorts IAR compile warnings about conversion from enum to int and vice versa. This doesn't handle the case in #7917 which requires further discussion.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required not a functionality change
- [x] **backport** not required as above
- [x] **tests** should be covered

### CODE REVIEW

**Review by daverodgman — DISMISSED** [review]

File: tests/src/psa_exercise_key.c [hunk 1]
```diff
@@ -692,7 +692,7 @@ int mbedtls_test_psa_exported_key_sanity_check(
         TEST_EQUAL(mbedtls_asn1_get_tag(&p, end, &len,
                                         MBEDTLS_ASN1_SEQUENCE |
                                         MBEDTLS_ASN1_CONSTRUCTED), 0);
-        TEST_EQUAL(len, end - p);
+        TEST_EQUAL(len, (uintptr_t) end - (uintptr_t) p);
```
Comment (daverodgman) [inline tests/src/psa_exercise_key.c:695]: I'm not sure this is the right way to do it. AIUI, `end - p` has type `ptrdiff_t`, i.e., signed, so I think you should be able to write as `(size_t)(end - p)`

Comment (AgathiyanB) [reply tests/src/psa_exercise_key.c]: After this I had put casts in the TEST_EQUAL macro so in fact I can remove the casts on this line entirely

**Review by daverodgman — DISMISSED** [review]

File: tests/src/psa_exercise_key.c (hunk 1 above)
Comment (daverodgman) [reply tests/src/psa_exercise_key.c]: Great, that's the only bit I would have wanted a second reviewer for so I'm now marking single-reviewer.

**Review by daverodgman — APPROVED** [review]

**Comment by daverodgman** [comment]
@gilles-peskine-arm not sure if you are reviewing or not? If not I'll merge this as single-reviewer.

**Comment by gilles-peskine-arm** [comment]
I'm not reviewing, I was just looking at the issue for which you wanted a second reviewer and noticed that it wasn't completely resolved. I'm happy to merge this as single-reviewer and haven't done an actual review.

**Pull request merged**